driver-core/src/main/com/mongodb/internal/client/model/AbstractConstructibleBson.java

```diff
@@ -70,27 +81,39 @@ public final boolean equals(final Object o) {
             return false;
         }
         final AbstractConstructibleBson<?> that = (AbstractConstructibleBson<?>) o;
-        return doc.equals(that.doc);
+        return Objects.equals(base, that.base) && Objects.equals(appended, that.appended);
     }
 
     @Override
     public final int hashCode() {
-        return Objects.hash(doc);
+        return Objects.hash(base, appended);
     }
 
     @Override
     public String toString() {
-        return doc.toString();
+        return "{base=" + base
+                + ", appended=" + appended
+                + '}';
+    }
+
+    private static BsonDocument newAppended(final BsonDocument base, final BsonDocument appended) {
```

Comment:
I initially got confused by the existence of a `newAppended` non-static method and this new static one.  Can you think of a different name for one of them?  They seem semantically different so this kind-of overload is confusing.

Reply:
Renamed to `newMerged`.